src/utils/generateLinkToShare.ts

```diff
@@ -0,0 +1,9 @@
+import { ResponseType } from "~/types";
+
+export function encodeStateToUrl(originalContent: string, responseObject: ResponseType): string {
+  const baseUrl = (process.env.BASE_URL || window.location.origin) + "/result";
```

Comment:
Really cool!

So about the `/results` page I think it's fine, but couldn't you just render the states based on whether or not a query parameter exists in the url or would that need client hydration to detect?

Reply:
@MaiaS Adding comments here too about the things we discussed on the call for ref:
Since we are using adding the pages inside App directory, we will have to extract the query params through the `searchParams` props of the Page Component. And this can be only done if the page.tsx is server side rendered. For this, had to separate out the client side code inside page.tsx into a separate file (`clientPage.tsx`).  

Also, here is the link to an open issue related to searchParams: https://github.com/vercel/next.js/issues/43077